server.go

```diff
@@ -602,13 +752,18 @@ func (s *Server) Stop() {
 	s.lis = nil
 	cs := s.conns
 	s.conns = nil
+	for c := range s.hconns {
+		go c.Close()
```

Comment:
I would prefer the same treatment like s.conns. It does not matter how long it takes Stop to return as long as it does not hold the lock.

We do need a graceful shutdown (issue#147). But Stop() is not a method for testing only. Some service providers may want to stop a grpc server immediately but they do not want to crash/exit the entire process (there are other modules working for other purposes).